cpp/src/arrow/compute/kernels/aggregate_benchmark.cc

```diff
@@ -398,5 +398,17 @@ MINMAX_KERNEL_BENCHMARK(MinMaxKernelInt16, Int16Type);
 MINMAX_KERNEL_BENCHMARK(MinMaxKernelInt32, Int32Type);
 MINMAX_KERNEL_BENCHMARK(MinMaxKernelInt64, Int64Type);
 
+static void CountKernelBenchInt64(benchmark::State& state) {
+  RegressionArgs args(state);
+  const int64_t array_size = args.size / sizeof(int64_t);
+  auto rand = random::RandomArrayGenerator(1923);
+  auto array = rand.Numeric<Int64Type>(array_size, -100, 100, args.null_proportion);
+
+  for (auto _ : state) {
+    ABORT_NOT_OK(Count(array).status());
```

Comment:
The null count is only computed once and then cached on the array, so this benchmark isn't correct.

Reply:
Indeed it's computed again every time. The exec engine runs on batch iteration though for this case only one batch actually occurs. Each iteration use Slice to fetch the part of input array.
See https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/exec.cc#L174

Reply:
Should I change this benchmark to use Slice in case the exec engine optimize the iteration?
ABORT_NOT_OK(Count(array->Slice(0, array_size)).status());

Reply:
Just add a path to Slice if it's a exactly copy of original. Thanks.